### PR TITLE
Add safer check for showing download progress bar

### LINKED ISSUE #13351: IERS update fails when logging to non-standard stdout
### Description

I'm using a custom class to redirect stdout to a logger, I believe it's taken from this Stack Overflow answer (https://stackoverflow.com/a/39215961).

Because I wasn't inheriting from an IO base class the `sys.stdout` did not have `isatty` defined, which caused an error in the code which checks for IERS updates:

https://github.com/astropy/astropy/blob/7eb4cf08c82ff5b084112792ccda68e11705dace/astropy/utils/data.py#L1182-L1185

The `isatty` check was added in https://github.com/astropy/astropy/commit/b68ec5da02ab59bde58eb4cc36597b71a822a6fc in order to prevent loading bars being shown when not in a terminal (#7577).

The only reason I'm bringing this here is that I came across https://github.com/ipython/ipython/issues/9832 when searching, which means I'm not the only person who's had a similar case (almost certainly having copy-pasted the same code from SO). The fix is really simple to check `hasattr(sys.stdout, 'isatty')` (as implemented in https://github.com/ipython/ipython/pull/9833/commits/3e16cbc4bf3305f26ec7a7db53931d6c87ea2213), and while not strictly Astropy's problem I thought it was worth raising here just in case anyone has the same issue in the future. (For the benefit of anyone similarly searching: the fix for my case is to inherit from `io.TextIOBase`, which I found in the comments on SO).

### Expected behavior

No errors.

### Actual behavior

When running astropy code which triggers the IERS download I receive the following warning:
```python
failed to download ftp://anonymous:mail%40astropy.org@gdc.cddis.eosdis.nasa.gov/pub/products/iers/finals2000A.all and https://datacenter.iers.org/data/9/finals2000A.all:
    'StreamToLogger' object has no attribute 'isatty'.
```
This causes the IERS table to not be updated, which I only found out 30+ days later when I started seeing errors with `interpolating from IERS_Auto using predictive values that are more
than 30.0 days old`.

### PR DESCRIPTION
**Pull request by martinjohndyer**
### Description

This pull request is to address issue https://github.com/astropy/astropy/issues/13351. Basically users can overwrite `sys.stdout` with custom stream classes which might not have the `isatty` method (see that issue for an example), leading the check added in #7577 to fail. It's safer to check using `hasattr` first, then if it's missing then assume we're not in a terminal and therefore don't want to show the loading bar.

Fixes #13351

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Maybe we don't need a test but can you please add a change log under bugfix? Thanks!

https://github.com/astropy/astropy/blob/main/docs/changes/README.rst

**Comment by martinjohndyer** [comment]
> Maybe we don't need a test but can you please add a change log under bugfix? Thanks!
> 
> [https://github.com/astropy/astropy/blob/main/docs/changes/README.rst](https://github.com/astropy/astropy/blob/main/docs/changes/README.rst?rgh-link-date=2022-06-17T16%3A27%3A34Z)

Changelog added. It seems #7577 didn't add any tests, writing something that involves altering `stdout` sounds awkward so I hope it's not necessary.

**Review by pllim — APPROVED** [review]
I agree w.r.t. not needing test as long as you already tested this branch locally with your use case. The patch seems straightforward enough. LGTM. Thanks!

File: docs/changes/utils/13352.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix error when supressing download progress bar while using non-default
```
Comment (pllim) [inline docs/changes/utils/13352.bugfix.rst:1]: Oops, a typo.
```suggestion
Fix error when suppressing download progress bar while using non-default
```

Comment (martinjohndyer) [reply docs/changes/utils/13352.bugfix.rst:1]: D'oh! That's what you get when submitting last thing on a Friday :flushed:

**Comment by pllim** [comment]
dev job failures are unrelated, so merging. Thanks again!

**Pull request merged**